test/gst-va/vpp/deinterlace.py

```diff
@@ -0,0 +1,107 @@
+###
+### Copyright (C) 2023 Intel Corporation
+###
+### SPDX-License-Identifier: BSD-3-Clause
+###
+
+from ....lib import *
+from ....lib.gstreamer.va.util import *
+from ....lib.gstreamer.va.vpp import VppTest
+
+if len(load_test_spec("vpp", "deinterlace")):
+  slash.logger.warn(
+    "gst-va: vpp deinterlace with raw input is no longer supported")
+
+@slash.requires(*platform.have_caps("vpp", "deinterlace"))
+class DeinterlaceTest(VppTest):
+  _default_methods_ = [
+    "bob",
+    "motion-adaptive",
+    "motion-compensated",
+  ]
+
+  _default_modes_ = [
+    dict(method = m, rate = "field") for m in _default_methods_
+  ]
+
+  def before(self):
+    vars(self).update(
+      metric  = dict(type = "md5"), # default metric
+      vpp_op  = "deinterlace",
+    )
+    super(DeinterlaceTest, self).before()
+
+  def init(self, tspec, case, method, rate):
+    vars(self).update(tspec[case].copy())
+    vars(self).update(case = case, method = method, rate = rate)
+
+    self.gstdecoder = self.gstdecoder.format(**vars(self))
+    self.frames *= 2 # field rate produces double number of frames
+
+  def validate_caps(self):
+    self.caps = platform.get_caps(
+      "vpp", "deinterlace", self.method.replace('-', '_'))
+
+    if self.caps is None:
+      slash.skip_test(
+        format_value(
+          "{platform}.{driver}.{method} not supported", **vars(self)))
+
+    # The rate is fixed in vaapipostproc deinterlace.  It always outputs at
+    # field rate (one frame of output for each field).
```

Comment:
Please update this comment... "vaapipostproc" is not part of this context.

Reply:
updated